solutions/ModernSearch/react-search-refiners/spfx/src/services/SearchService/SearchService.ts

```diff
@@ -383,6 +388,60 @@ class SearchService implements ISearchService {
 
         return refinementQueryConditions;
     }
+
+    // Function to inject synonyms at run-time
+    private _injectSynonyms(query: string): string {
+        // Remove complex query parts AND/OR/NOT/ANY/ALL/parenthasis/property queries/exclusions - can probably be improved            
+        const cleanQuery = query.replace(/(-\w+)|(-"\w+.*?")|(-?\w+[:=<>]+\w+)|(-?\w+[:=<>]+".*?")|((\w+)?\(.*?\))|(AND)|(OR)|(NOT)/g, '');
+        const queryParts: string[] = cleanQuery.match(/("[^"]+"|[^"\s]+)/g);
+
+        // code which should modify the current query based on context for each new query
+        if (queryParts && this._synonymTable) {
```

Comment:
Hi @aboredman , can you move the table check to line 395 and check for null, and size of 0. Saves doing regex on the query if there are no synonyms. Minor, but better 🙂